lib/layer/core.py

```diff
@@ -237,6 +238,20 @@ def clear(self):
     ## Properties
 
     @property
+    def group(self):
```

Comment:
Please add some doctests (like `root`'s) for this property. In particular, the doctest _is_ documentation, so it should readably test which group of a nested structure a layer's `group` property refers to.

![image](https://cloud.githubusercontent.com/assets/61299/15800966/4c883c58-2a7f-11e6-9228-f7911aca9e28.png)

We need to be pretty fanatical about CI checks now the project is growing.